js/private/lifecycle/lifecycle-hooks.js

```diff
@@ -0,0 +1,78 @@
+const {
+    mkdirSync,
+    copyFileSync,
+    readdirSync,
+    readFileSync,
+    existsSync,
+    statSync,
+} = require('fs')
+const path = require('path')
+const { runPostinstallHooks } = require('@pnpm/lifecycle')
+
+async function main(argv) {
+    if (argv.length !== 2) {
+        console.error('Usage: node lifecycle-hooks.js [packageDir] [outputDir]')
+        process.exit(1)
+    }
+    const packageDir = argv[0]
+    const outputDir = argv[1]
+
+    copyPackageContents(packageDir, outputDir)
+
+    const packageJson = JSON.parse(
+        readFileSync(path.join(packageDir, 'package.json'))
+    )
+
+    if (
+        packageJson.scripts?.preinstall ||
+        packageJson.scripts?.install ||
+        packageJson.scripts?.postinstall
+    ) {
+        const nodeModulesPath = path.join(outputDir, '..', 'node_modules')
```

Comment:
how does this pathing work out that node_modules is `$(outputDir)/../node_modules`

Reply:
TODO: Leave comment here explaining